src/js/popup.js

```diff
@@ -89,44 +99,58 @@ $(function () {
 
         var combinedTabs = [];
 
-        // Build buckets of tabs keyed by hostname
-        _.forEach(_.keys(WINDOW_TABS), function (windowId) {
+        // Combine unpinned tabs for all windows, ordered by `windowId`
+        _.forEach(_.sortedUniq(_.keys(WINDOW_TABS)), function (windowId) {
             var tabs = WINDOW_TABS[windowId];
             _.forEach(tabs, function (tab) {
                 var match = HOST_REGEX.exec(tab.url);
                 if (match) {
+                    // include tabs with actual webpages
                     combinedTabs.push(tab);
                 } else {
                     // skip non-matching URLS, e.g. chrome://newtab/
                 }
             });
         });
 
-        // Sort tabs by URL
-        var orderedTabs = _.sortBy(combinedTabs, [
+        // Sort unpinned tabs by URL
+        var orderedCombinedTabs = _.sortBy(combinedTabs, [
             function (tab) {
                 return tab.url;
             }
         ]);
 
+        // Combine pinned tabs with unpinned tabs
+        var orderedTabs = _.concat(PINNED_TABS, orderedCombinedTabs);
+
         // Create a new window and move tabs over
-        moveTabsToNewWindow(orderedTabs, closeBlankTabs);
+        moveTabsToNewWindow(orderedTabs)
+            .then(function (movedTabs) {
+                return closeBlankTabs();
+            })
+            .then(function (closedTabs) {
+                repinTabs();
+            });
     }
 
-    function moveTabsToNewWindow(tabs, callback) {
+    function moveTabsToNewWindow(tabs) {
         var tabIds = _.map(tabs, function (tab) {
             return tab.id;
         });
 
         // https://developer.chrome.com/extensions/windows#method-create
         var createData = {};
-        chrome.windows.create(createData, function (window) {
+        return chrome.windows.create(createData).then(function (window) {
             // https://developer.chrome.com/extensions/tabs#method-move
             var moveProperties = {
                 windowId: window.id,
                 index: -1
             };
-            chrome.tabs.move(tabIds, moveProperties, callback);
+
+            // Open bug in Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=876460&q=chrome.tabs.move&can=2
```

Comment:
Tracking this bug here: https://bugs.chromium.org/p/chromium/issues/detail?id=876460&q=chrome.tabs.move&can=2